back port additional changes to release/202502 [REBASE&FF]

apop5 opened the pull request:
## Description

Backporting the changes for 
b7a26a3198ea5998bde35cf565ac3472d0153081
ed338da623774d34f10b48a47e8305f127afa4c9
e675273ce6a20e412b622036513ec7db66965192
ca48fd205ac7c3fb9f6fe033174d8b4a565c1467
e4325d5bcf8253a8a44fcea18330065e9cd542af
b7712d33c5d55e0f53fce58372c7567fbff08b6a
694ba3b79cee4c36e80fa3e1d27169c673a043b4

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [ ] Backport to release branch?

## How This Was Tested
Local CI

## Integration Instructions
No integration necessary

VivianNK committed:
BaseTools: Update tools_def for aarch64

/WX was added as a build flag and now aarch64 builds
fail because of an undocumented linker warning: 4226 Alignment specified
exceeds target machine page size.
This linker warning has always occured and can be ignored.

Signed-off-by: Vivian Nowka-Keane <vnowkakeane@linux.microsoft.com>

VivianNK committed:
BaseTools: Change arm align value from 32 to 64

Alignment needed to be updated to 64 because of a linker warning
when building with MSVC.
/ALIGN:64 is the minimum alignment for MSVC ARM64, which differs from
MSVC x64. This was missed when checking into edk2 because CI isn't
run for MSVC ARM64.

Signed-off-by: Vivian Nowka-Keane <vnowkakeane@linux.microsoft.com>

kuqin12 committed:
MdeModulePkg: PiSmmIpl: Fix physical address dereferencing

With current implemenation, all 3 SmmCommunication* functions go through
the same routine, which will dereference the incoming pointer to inspect
whether this is a V3 buffer or not.

However, the caller always pass in the physical addresses, which could
cause the system to page fault after OS take over the runtime control.

This change reverted the common routine to its previous form to handle MM
communicate v1 and v2. Additionally, a specific communicate function for
v3 was created to support MM communicate v3.

Co-authored-by: Michael D Kinney <michael.d.kinney@intel.com>
Signed-off-by: Kun Qin <kun.qin@microsoft.com>

kuqin12 commented:
I thought the backport checkbox works?

VivianNK commented:
> I thought the backport checkbox works?

I didn't realize the release branch was up so I didn't check backporting

kenlautner committed:
Revert "Implement MmServicesTableLib for MM Core (#730)"

This reverts commit 26147980f53dc0db8084a617ce58460e92622699.

kenlautner committed:
Remove component that no longer exists.

apop5 committed:
Add BuildReportEvaluation post build plugin (#1297)

## Description

Add a post build plugin that will parse the build report and run
evaluation queries over it.

Infrastructure allows for platforms to register additional queries,
either under the existing categories or creating their own.

See BuildReportEvaluationReadMe.md for more details.


- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [ ] Backport to release branch?

## How This Was Tested
Ran on virtual platforms builds with scope enabled. 
Ran on physical platform build with scope enabled. 

## Integration Instructions
Add platform-checks scope.
Ensure that BUILDREPORT_FILE is enabled and configured.

apop5 committed:
[Squash on Rebase] Update release-basetools to use newer ubuntu. (#1343)

## Description

Ubuntu 20.04 on github have been deprecated and
have stopped working as of 2025.04.15.

Update to use ubuntu 22.04.

https://github.com/actions/runner-images/issues/11101


- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [ ] Backport to release branch?

## How This Was Tested
CI running of pipeline

## Integration Instructions
no integration necessary.
